ui/src/views/image/UpdateTemplate.vue

```diff
@@ -0,0 +1,389 @@
+          description: 'osdefault'
+        })
+        controller.push({
+          id: 'scsi',
+          description: 'scsi'
+        })
+        controller.push({
+          id: 'virtio',
+          description: 'virtio'
+        })
+      } else if (hyperVisor === 'VMware') {
+        controller.push({
+          id: '',
+          description: ''
+        })
+        controller.push({
+          id: 'scsi',
+          description: 'scsi'
+        })
+        controller.push({
+          id: 'ide',
+          description: 'ide'
+        })
+        controller.push({
+          id: 'osdefault',
+          description: 'osdefault'
+        })
+        controller.push({
+          id: 'pvscsi',
+          description: 'pvscsi'
+        })
+        controller.push({
+          id: 'lsilogic',
+          description: 'lsilogic'
+        })
+        controller.push({
+          id: 'lsisas1068',
+          description: 'lsilogicsas'
+        })
+        controller.push({
+          id: 'buslogic',
+          description: 'buslogic'
+        })
+      }
+
+      this.$set(this.rootDisk, 'opts', controller)
+    },
+    fetchNicAdapterTypes () {
+      const nicAdapterType = []
+      nicAdapterType.push({
+        id: '',
+        description: ''
+      })
+      nicAdapterType.push({
+        id: 'E1000',
+        description: 'E1000'
+      })
+      nicAdapterType.push({
+        id: 'PCNet32',
+        description: 'PCNet32'
+      })
+      nicAdapterType.push({
+        id: 'Vmxnet2',
+        description: 'Vmxnet2'
+      })
+      nicAdapterType.push({
+        id: 'Vmxnet3',
+        description: 'Vmxnet3'
+      })
+
+      this.$set(this.nicAdapterType, 'opts', nicAdapterType)
+    },
+    fetchKeyboardTypes () {
+      const keyboardType = []
+      const keyboardOpts = this.$config.keyboardOptions || {}
+      keyboardType.push({
+        id: '',
+        description: ''
+      })
+
+      Object.keys(keyboardOpts).forEach(keyboard => {
+        keyboardType.push({
+          id: keyboard,
+          description: this.$t(keyboardOpts[keyboard])
+        })
+      })
+
+      this.$set(this.keyboardType, 'opts', keyboardType)
+    },
+    handleSubmit (e) {
+      e.preventDefault()
+      if (this.loading) return
+      this.form.validateFields((err, values) => {
+        if (err) {
+          return
+        }
+        this.loading = true
+        const params = {
+          id: this.resource.id
+        }
+        const detailsField = ['rootDiskController', 'nicAdapter', 'keyboardType']
+        for (const key in values) {
+          if (!this.isValidValueForKey(values, key)) continue
+          if (detailsField.includes(key)) {
+            params['details[0].' + key] = values[key]
+            continue
+          }
+          params[key] = values[key]
+        }
+        api('updateTemplate', params).then(json => {
+          this.$message.success(`${this.$t('message.success.update.template')}: ${this.resource.name}`)
```

Comment:
@shwstppr Can you add the translate this key.
![image](https://user-images.githubusercontent.com/13766648/149719271-8cf583f7-42d2-4e97-a411-064b93540ca9.png)